### DISCUSSION TITLE
kluctl vs helmfile

### DISCUSSION THREAD

**Discussion by lucasfcnunes** [helmfile/helmfile#1009] [General]
It's already applied to CNCF Sandbox https://github.com/cncf/sandbox/issues/8. Could helmfile do the same?

[kluctl on helmfile](https://kluctl.io/blog/2023/02/28/managing-many-helm-charts-with-kluctl/#helmfile):
> "[Helmfile](https://helmfile.readthedocs.io/en/latest/) is probably the tool that can be best compared to Kluctl. There are however some important differences. Helmfile concentrates on Helm as being the underlying building block for all deployments. This means, it will internally convert plain manifests and Kustomize deployments into Helm Charts and install them via the Helm CLI.

> Kluctl on the other hand uses Kustomize as the base building block and instead converts Helm Charts into Kustomize deployments by simply rendering the Charts. This also means, that Kluctl will not use helm install but instead rely on its own server-side-apply based implementation, giving it much better conflict detection, conflict resolution and diffing capabilities.

> Helmfile currently has no GitOps style controller available. From what I can find online, people tend to use helmfile template to simply generate plain manifests which they then commit and push to Git."

Using the kustomize approach instead of the helm one could be more beneficial?
About GitOps I too think dependabot compatibility, `helmfile list` rich env/kubeContext/release/version info, etc could be improved.

Other:
- https://github.com/kluctl/kluctl/issues/15

**Comment by yxxhero** [comment]
@mumoshu WDYT?

**Reply by pathob** [reply]
I'm not mumoshu but in my opinion it would be great if Helmfile would join a foundation.

**Reply by yxxhero** [reply]
@pathob I think so.

**Reply by lucasfcnunes** [reply]
@yxxhero, what does @mumoshu think about this?
